gammapy/background/reflected.py

```diff
@@ -190,6 +197,9 @@ def find_regions(self):
                 curr_angle = curr_angle + self.angle_increment
 
         log.debug('Found {} reflected regions'.format(len(reflected_regions)))
+        if self.max_region_number is not None:
```

Comment:
Isn't it easier to move this check up into the while loop, and `break` if the max is reached?
Should be one line less, and a bit faster, no?